pkg/apis/eksctl.io/v1alpha4/types.go

```diff
@@ -336,8 +351,12 @@ func (c *ClusterConfig) NewNodeGroup() *NodeGroup {
 			WithLocal:  Enabled(),
 			WithShared: Enabled(),
 		},
-		DesiredCapacity: nil,
+		DesiredCapacity: NewInt(DefaultNodeCount),
```

Comment:
So actually having no default size is a feature, which is to do with cluster autoscaler. I believe the idea is that user should be able to set just one of the 3 parameters.

Reply:
defaults are the only sane way to write this kind of things, otherwise you'll get tons of subtle bugs every time you want to change something (as I've experienced while working on this PR)